src/shared/Microsoft.AzureRepos/AzureReposHostProvider.cs

```diff
@@ -90,28 +151,97 @@ public override async Task<ICredential> GenerateCredentialAsync(InputArguments i
                 remoteUri,
                 null);
             string atUser = accessToken.GetAzureUserName();
-            Context.Trace.WriteLineSecrets($"Acquired Azure access token. User='{atUser}' Token='{{0}}'", new object[] {accessToken.EncodedToken});
+            _context.Trace.WriteLineSecrets($"Acquired Azure access token. User='{atUser}' Token='{{0}}'", new object[] {accessToken.EncodedToken});
 
             // Ask the Azure DevOps instance to create a new PAT
             var patScopes = new[]
             {
                 AzureDevOpsConstants.PersonalAccessTokenScopes.ReposWrite,
                 AzureDevOpsConstants.PersonalAccessTokenScopes.ArtifactsRead
             };
-            Context.Trace.WriteLine($"Creating Azure DevOps PAT with scopes '{string.Join(", ", patScopes)}'...");
+            _context.Trace.WriteLine($"Creating Azure DevOps PAT with scopes '{string.Join(", ", patScopes)}'...");
             string pat = await _azDevOps.CreatePersonalAccessTokenAsync(
                 orgUri,
                 accessToken,
                 patScopes);
-            Context.Trace.WriteLineSecrets("PAT created. PAT='{0}'", new object[] {pat});
+            _context.Trace.WriteLineSecrets("PAT created. PAT='{0}'", new object[] {pat});
 
             return new GitCredential(atUser, pat);
         }
 
-        protected override void ReleaseManagedResources()
+        /// <remarks>
+        /// For dev.azure.com-style URLs we use the path arg to get the Azure DevOps organization name.
+        /// We ensure the presence of the path arg by setting credential.useHttpPath = true at install time.
+        ///
+        /// The result of this workaround is that we are now unable to determine if the user wanted to store
+        /// credentials with the full path or not for dev.azure.com-style URLs.
+        ///
+        /// Rather than always assume we're storing credentials against the full path, and therefore resulting
+        /// in an personal access token being created per remote URL/repository, we never store against
+        /// the full path and always store with the organization URL "dev.azure.com/org".
```

Comment:
So we still require `credential.useHttpPath`, but we are dropping _some_ data from the URL?

Reply:
GCM Core has _always_ required `credential.useHttpPath` for `dev.azure.com` URLs, since we don't implement the same hackery of walking the process hierarchy and extracting the parent process' command lines like GCM for Windows does. Plus we cannot do this method at all on non-Windows systems (requires we run as `root`).

In this regard, GCM Core does not support the scenario of using the full path with `dev.azure.com` URLs whereas GCM4Windows does. I'd rather break compat here than introduce that code which when first introduced caused systems to crash due to memory leaks, infinite tree walking, and issues with PID reuse.

In 99% of cases, users of GCM4W will _not_ have `useHttpPath` set and will be storing against the org URL anyway.

The preferred fix here is to change Git itself so we can avoid forcing `useHttpPath` to true for `dev.azure.com`.